### PR TITLE
[CAS-555] Repo facade5

### PR DESCRIPTION
**Pull request by ogkuzmin**
### Description

Final step.
1) Rename `RepositoryHelper` to `RepositoryFacade`
2) Move instantiation logic to the brand new `RepositoryFacadeBuilder`
3) Move entities/dao/mappers/repos under repository package

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/repository/builder/RepositoryFactory.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import io.getstream.chat.android.livedata.repository.domain.queryChannels.QueryChannelsRepository
+import io.getstream.chat.android.livedata.repository.domain.queryChannels.QueryChannelsRepositoryImpl
+import io.getstream.chat.android.livedata.repository.domain.reaction.ReactionRepository
+import io.getstream.chat.android.livedata.repository.domain.reaction.ReactionRepositoryImpl
+import io.getstream.chat.android.livedata.repository.domain.syncState.SyncStateRepository
+import io.getstream.chat.android.livedata.repository.domain.syncState.SyncStateRepositoryImpl
+import io.getstream.chat.android.livedata.repository.domain.user.UserRepository
+import io.getstream.chat.android.livedata.repository.domain.user.UserRepositoryImpl
+
+internal class RepositoryFactory(
```
Comment (ogkuzmin) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/repository/builder/RepositoryFactory.kt:21]: Just moved from other place. No new code

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/repository/domain/syncState/SyncStateMapper.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package io.getstream.chat.android.livedata.repository.domain.syncState
+
+import io.getstream.chat.android.livedata.model.SyncState
+
+internal fun SyncStateEntity.toModel() =
```
Comment (ogkuzmin) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/repository/domain/syncState/SyncStateMapper.kt:5]: Also moved to new package. No new code

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/repository/helper/BaseRepositoryFacadeTest.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import io.getstream.chat.android.livedata.repository.domain.queryChannels.QueryChannelsRepository
+import io.getstream.chat.android.livedata.repository.domain.reaction.ReactionRepository
+import io.getstream.chat.android.livedata.repository.domain.syncState.SyncStateRepository
+import io.getstream.chat.android.livedata.repository.domain.user.UserRepository
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineScope
+import org.junit.jupiter.api.BeforeEach
+
+@ExperimentalCoroutinesApi
+internal open class BaseRepositoryFacadeTest {
```
Comment (ogkuzmin) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/repository/helper/BaseRepositoryFacadeTest.kt:18]: Rename from `BaseRepositoryHelperTest` to `BaseRepositoryFacadeTest`. No new code

**Review by adasiewiczr — APPROVED** [review]
Good job!

**Pull request merged**